salt/_modules/azurearm_network.py

```diff
@@ -2749,3 +2749,1408 @@ def route_tables_list_all(**kwargs):
+
+    :param name: The name of the virtual network gateway to reset.
+
+    :param resource_group: The name of the resource group.
+
+    :param gateway_vip: Virtual network gateway vip address supplied to the begin 
+        reset of the active-active feature enabled gateway.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-call azurearm_network.virtual_network_gateway_reset test_name test_group
+
+    '''
+    result = False
+    netconn = __utils__['azurearm.get_client']('network', **kwargs)
+    try:
+        reset = netconn.virtual_network_gateways.reset(
+            resource_group_name=resource_group,
+            virtual_network_gateway_name=name,
+            gateway_vip=gateway_vip
+        )
+        reset.wait()
+        result = True
+    except CloudError as exc:
+        __utils__['azurearm.log_cloud_error']('network', str(exc), **kwargs)
+        result = {'error': str(exc)}
+
+    return result
+
+
+def virtual_network_gateway_reset_vpn_client_shared_key(name, resource_group, **kwargs):
+    '''
+    .. versionadded:: Sodium
+
+    Resets the VPN client shared key of the virtual network gateway in the specified resource group.
+
+    :param name: The name of the virtual network gateway.
+
+    :param resource_group: The name of the resource group.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-call azurearm_network.virtual_network_gateway_reset_vpn_client_shared_key test_name test_group 
+
+    '''
+    result = {}
+    netconn = __utils__['azurearm.get_client']('network', **kwargs)
+    try:
+        reset = netconn.virtual_network_gateways.reset_vpn_client_shared_key(
+            resource_group_name=resource_group,
+            virtual_network_gateway_name=name
+        )
+
+        reset_result = reset.result()
+        result = reset_result.as_dict()
+    except CloudError as exc:
+        __utils__['azurearm.log_cloud_error']('network', str(exc), **kwargs)
+        result = {'error': str(exc)}
+
+    return result
+
+
+def virtual_network_gateway_generatevpnclientpackage(name, resource_group, processor_architecture, 
+                                                     authentication_method, radius_server_auth_certificate,
+                                                     client_root_certificates, **kwargs):
+    '''
+    .. versionadded:: Sodium
+
+    Generates VPN client package for P2S client of the virtual network 
+        gateway in the specified resource group.
+
+    :param name: The name of the virtual network gateway.
+
+    :param resource_group: The name of the resource group.
+
+    :param processor_architecture: VPN client Processor Architecture. Possible values include: 'Amd64', 'X86'
+
+    :param authentication_method: VPN client authentication method. Possible values include: 'EAPTLS', 'EAPMSCHAPv2'
+
+    :param radius_server_auth_certificate: The public certificate data for the radius server authentication
```

Comment:
Make `radius_server_auth_certificate` and `client_root_certificates` default to `None` (optional args). If they don't read the notes and don't provide it in the correct cases, then that's on them.